guidelines/README.md

```diff
@@ -0,0 +1,462 @@
+# Runtastic Flow Guidelines
+
+## Table of Contents
+  0. [Preface](#preface)
+  1. [Type Names](#type-names)
+  2. [Type Definitions](#type-definitions)
+  3. [Function Headers](#function-headers)
+  4. [Type Inference](#type-inference)
+  5. [Libdef Files](#libdef-files)
+
+## Preface
+
+This document discusses our ways of writing flowtyped JavaScript source code
+for Runtastic related projects.
+
+To stay consistent between examples, it is assumed that source code is located
+in `[PROJECT]/src/` and its transpiled output in `[PROJECT]/lib/`. Paths may
+vary project-wise, but we try to stick to this convention as best as possible.
+
+**[⬆ back to top](#table-of-contents)**
+
+## Type Names
+
+Type names are a very important part of documentation, so it must be clear what
+intentions each type has, e.g. a consumer of a type should intuitively know what
+meta-type (`Object`, `Function` or primitives like `string`, `number`,...) they
+represent. Also names must not be ambiguous to prevent mistaking JavaScript
+with Flow code.
+
+  <a name="type-names--capitalize"></a><a name="1.1"></a>
+  - [1.1](#type-names--capitalize) Capitalize & CamelCase type names
+
+  > Why? This will help us prevent collisions with variable names etc.
+
+  ```javascript
+  // bad
+  type foo = string;
+  type myFoo = string;
+  type JSONApiEntity = {};
+
+  // good
+  type Foo = string;
+  type MyFoo = string;
+  type JsonApiEntity = {};
```

Comment:
TBH I'm usually in favor of writing acronyms in uppercase, so `JSONAPIEntity` in this case, but in this specific case it really looks weird but still it is more human readable.

Acronyms not written in uppercase can end up with a completely different meaning.
for instance, you want to load the members of the Austrian League of Legends (ALL - fictional):
`loadAllMembers` vs. `loadALLMembers`